apps/meeting/src/components/DeviceSelection/CameraDevices/BackgroundReplacementDropdown.tsx

```diff
@@ -0,0 +1,79 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+import React, { ChangeEvent, useState } from 'react';
+import {
+  useBackgroundReplacement,
+  FormField,
+  Select,
+  useVideoInputs,
+} from 'amazon-chime-sdk-component-library-react';
+import { BackgroundImageEncoding, createColorBlob, createImageBlob } from '../../../utils/image';
+import { ReplacementDropdownOptionType, ReplacementOptions } from '../../../types/index';
+import { useAppState } from '../../../providers/AppStateProvider';
+
+interface Props {
+  /* Title for the dropdown, defaults to `Background Replacement Dropdown` */
+  label?: string;
+}
+
+export const BackgroundReplacementDropdown: React.FC<Props> = ({
+  label = 'Background Replacement Dropdown',
+}) => {
+  const { selectedDevice } = useVideoInputs();
+  const { backgroundReplacementOption, setBackgroundReplacementOption } = useAppState();
+  const { isBackgroundReplacementSupported, changeBackgroundReplacementImage } = useBackgroundReplacement();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const options: ReplacementDropdownOptionType[] = [
+    {
+      label: ReplacementOptions.Blue,
+      value: isBackgroundReplacementSupported === undefined || isBackgroundReplacementSupported === false ? 'Background Replacement not supported' :ReplacementOptions.Blue,
+    },
+    {
+      label: ReplacementOptions.Beach,
+      value: isBackgroundReplacementSupported === undefined || isBackgroundReplacementSupported === false ? 'Background Replacement not supported' :ReplacementOptions.Beach,
+    },
+  ];
+
+  // Creates a image blob on the selections (Blue, Beach) and changes the background image.
+  const selectReplacement = async (e: ChangeEvent<HTMLSelectElement>) => {
+    const selectReplacement = e.target.value;
+    let currentDevice = selectedDevice;
+
+    if (isLoading || currentDevice === undefined) {
+      return;
+    }
+    try {
+      setIsLoading(true);
+      if (selectReplacement === ReplacementOptions.Blue && isBackgroundReplacementSupported) {
+        const blob = await createColorBlob();
+        await changeBackgroundReplacementImage(blob);
+        console.info(`Video filter changed to Relacement - Blue`);
+      } else if (selectReplacement === ReplacementOptions.Beach && isBackgroundReplacementSupported) {
+        const imageInBase64 = BackgroundImageEncoding();
+        const blob = await createImageBlob(imageInBase64);
+        await changeBackgroundReplacementImage(blob);
+        console.info(`Video filter changed to Relacement - Beach`);
+      }      
```

Comment:
Could we get rid of these if else conditions and simply handles the logic based on an interface such as `options: ReplacementDropdownOptionType[]`. So we don't need to change this code to add any new options, we just need to expand the `options`

Reply:
If we do not handle the if-else logic here, we need to handle it in the createBlob, it will be the same case if we plan to add new options.

Reply:
Right, anyway we will need to handle it. I think it's better to handle it in a lowerlevel, such as createBlob.
I'd hope we can simplify the component layer interface to something like below:
```
const options = [
{
  label: 'Beach'
  type: 'image'
  value: 'image path or url'
 },
 {
   label: 'Blue',
   type: 'color',
   value: '#0000ff'
 }
]
```

And we when creating the blob, we can just check the type and create image blob or color blob separately.